build/OutputDirectories.props

```diff
@@ -25,5 +25,15 @@
       <TestPackagesDir>$(TestOutputDir)/packages/</TestPackagesDir>
       <TestArtifactsDir>$(TestOutputDir)/artifacts/</TestArtifactsDir>
       <TestResultXmlDir>$(TestOutputDir)/results/</TestResultXmlDir>
+      <ArchiveOutputDirectory>$(PackagesDirectory)</ArchiveOutputDirectory>
```

Comment:
> First we'll go ahead and finish the work to get properties out of targets

When you say "targets", do you mean .targets files or <Target> nodes?  I'm fine with moving them out of <Target> nodes as a rule/guideline.  I'm not fine with moving them out of .targets files as a rule/guidline.  They should only be moved into .props files if they are shared between multiple files.